bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/LedgerDirsMonitor.java

```diff
@@ -40,117 +45,109 @@
 /**
  * Thread to monitor the disk space periodically.
  */
-class LedgerDirsMonitor extends BookieThread {
+class LedgerDirsMonitor {
```

Comment:
I understand you are trying to use an executor for this, which is nice, but does it have to do anything with introducing mock executor? If not, this being a different commit will help reviewer.

Reply:
@jvrao : 

the PR is more about introducing mock executor to improve testing ability. 

I am using changing LedgerDirsMonitor from a thread based approach to an executor based approach as an example, to demonstrate how an executor based approach together with a mock executor can improve testing ability.

I was thinking of separating this PR into two, but that means the reviewers don't have the direct impression on how an executor based approach together with a mock executor can improve testing ability. I choose LedgerDirsMonitor as the example to demonstrate is becase it is a real self-contained class that can be easily converted from a thread based approach to an executor based approach.